magda-sleuther-format-layerer/src/onRecordFound.ts

```diff
@@ -0,0 +1,78 @@
+import * as _ from "lodash";
+
+import Registry from "@magda/typescript-common/dist/registry/AuthorizedRegistryClient";
+import { Record } from "@magda/typescript-common/dist/generated/registry/api";
+import { RecordLayer } from "@magda/typescript-common/src/registry-manual/api";
+import formatAspectDef, { FormatAspect } from "./formatAspectDef";
+import unionToThrowable from "@magda/typescript-common/src/util/unionToThrowable";
+
+import {
+    DcatFormatMeasure,
+    DownloadMeasure,
+    DownloadExtensionMeasure
+} from "./format-engine/measures";
+import { FormatSnapshotJudge } from "./format-engine/FormatSnapshotEvaluator";
+import { Snapshot } from "../../magda-typescript-common/src/format/MeasureSnapShot";
+
+export default async function onRecordFound(
+    record: Record,
+    registry: Registry
+) {
+    const distributions: Record[] =
+        record.aspects["dataset-distributions"] &&
+        record.aspects["dataset-distributions"].distributions;
+
+    if (!distributions || distributions.length === 0) {
+        return Promise.resolve();
+    }
+
+    const measureBundles: MeasureBundle[] = distributions.map(function(
+        distribution
+    ) {
+        return {
+            downloadMeasure: new DownloadMeasure(distribution),
+            extensionMeasure: new DownloadExtensionMeasure(distribution),
+            dcatMeasure: new DcatFormatMeasure(distribution)
+        };
+    });
+
+    const judgeResults: FormatSnapshotJudge[] = measureBundles.map(function(
+        measureBundle
+    ) {
+        return new FormatSnapshotJudge({
+            snapshots: [].concat(
```

Comment:
Rather than `[].concat( ... )` you can just put all 3 inside of `[ ... ]` right?

Reply:
4. Lastly you wrote to do [...] instead of [].concat. The problem is, I have 3 arrays and I want to concatenate them all into 1 array (flattened), if I put all 3 arrays inside [...], wont it return an array of arrays?

Reply:
oh right they're all arrays, no worries then :)